### PR TITLE
KAN-8 : SEARCH COMPONENT

### PR DESCRIPTION
**Pull request by GREEDnim**
## DESCRIPTION :
* ### Created ```LocationSearch``` Component, which can console the selected location via TWO ways.
> *   ### ```CurrentLocation``` Component to console the User's current location.
> *  ### ```LocationsContainer & Location ``` Component to display a list of locations .

* ### A DEBOUNCED SEARCH feature is also implemented when search for a specific location

## DEMO :

https://github.com/AYS-At-Your-Service/AYS/assets/84082104/14361826-a824-4834-8f15-97e212e96da6

### CODE REVIEW

File: Frontend/src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import './App.css'
-
+import  LocationSearch  from './components/LocationSearch/LocationSearch'
 function App() {
 
   return (
     <>
-      <h1>AYS</h1>
+      <LocationSearch></LocationSearch>
```
Comment (aravindv97) [inline Frontend/src/App.jsx:6]: pls remove

File: Frontend/src/App.jsx
Comment (aravind-digitech) [inline Frontend/src/App.jsx:1]: pls revert changes in this file

File: Frontend/src/assets/Data.js
Comment (aravind-digitech) [inline Frontend/src/assets/Data.js:1]: pls create a folder called services and name the file as location-data.service.js and use it

File: Frontend/src/components/CurrentLocation/CurrentLocation.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import currentLocation from '../../assets/currentLocation.svg'
+import './currentLocation.css'
+const CurrentLocation = () => {
+  const getCurrentCity = () => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
+    } else {
+      console.log("Geolocation is not supported by this browser.");
```
Comment (aravind-digitech) [inline Frontend/src/components/CurrentLocation/CurrentLocation.jsx]: pls remove all console logs

File: Frontend/src/components/CurrentLocation/CurrentLocation.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  const errorCallback = (error) => {
+    console.error("Error getting location:", error.message);
+  };
+
+  return (
+    <div className='current-location-container'  onClick={getCurrentCity}>
+      <img src={currentLocation} alt="" className='image' />
+      <div className="text">Use current location</div>
```
Comment (aravind-digitech) [inline Frontend/src/components/CurrentLocation/CurrentLocation.jsx]: pls save all constants in a separate constants file similar to https://github.com/AYS-At-Your-Service/AYS/pull/13/files#diff-c2c392d453cc978fcfed88aae84cdeb1b2ce75f60d0bb32f26cc6e821ab39a7b

File: Frontend/src/components/Location/Location.css [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.location-container{
+    padding:var(--location-container-padding);
+    display: flex;
+    gap:10px;
+}
+.location-container .image{
+    width:var(--location-image-width);
+    height: var(--location-image-width);
+    border-radius: 50%;
+    background-color:rgba(245,245,245,1.00);
```
Comment (aravind-digitech) [inline Frontend/src/components/Location/Location.css:14]: create variables for all numbers

File: Frontend/src/components/Location/Location.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import './Location.css'
+import locationImage from '../../assets/location.svg'
+
+export default function Location({data}){
+    return(
+        <>
+            <div className="location-container" onClick={()=>{console.log(data.main)}}>
```
Comment (aravind-digitech) [inline Frontend/src/components/Location/Location.jsx]: remove logs pls

File: Frontend/src/components/LocationSearch/LocationSearch.css [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+.modal-search-box-container .image{
+    width:var(--search-image-size);
+}
+.modal-search-box-container .text{
+    border: none;
+    outline: none;
+    flex-grow: 1;
+}
+.breaker-big{
+    border:5px solid rgb(234, 234, 234)
```
Comment (aravind-digitech) [inline Frontend/src/components/LocationSearch/LocationSearch.css:36]: use variable

File: Frontend/src/components/LocationsContainer/LocationsContainer.css [hunk 7]
```diff
@@ -0,0 +1,11 @@
+:root{
+    --location-container-height:400px;
```
Comment (aravind-digitech) [inline Frontend/src/components/LocationsContainer/LocationsContainer.css:2]: use variables

**Review by aravind-digitech — CHANGES_REQUESTED** [review]

File: Frontend/src/App.jsx
Comment (GREEDnim) [reply Frontend/src/App.jsx:1]: done

File: Frontend/src/assets/Data.js
Comment (GREEDnim) [reply Frontend/src/assets/Data.js:1]: done

File: Frontend/src/components/CurrentLocation/CurrentLocation.jsx (hunk 2 above)
Comment (GREEDnim) [reply Frontend/src/components/CurrentLocation/CurrentLocation.jsx]: done

File: Frontend/src/components/CurrentLocation/CurrentLocation.jsx (hunk 3 above)
Comment (GREEDnim) [reply Frontend/src/components/CurrentLocation/CurrentLocation.jsx]: done

File: Frontend/src/components/Location/Location.jsx (hunk 5 above)
Comment (GREEDnim) [reply Frontend/src/components/Location/Location.jsx]: done

File: Frontend/src/components/LocationsContainer/LocationsContainer.css (hunk 7 above)
Comment (GREEDnim) [reply Frontend/src/components/LocationsContainer/LocationsContainer.css:2]: done

**Review by Murali03M — APPROVED** [review]
done the review

**Review by Murali03M — APPROVED** [review]
done

**Review by aravind-digitech — APPROVED** [review]

**Review by Murali03M — APPROVED** [review]

**Pull request merged**